### PR TITLE
feat: add dead letter retry

### PR DESCRIPTION
**Pull request by nicoandrewss**
- adds exponential retry functionality using dead letter queues
- adds appropriate test
- adds some minor refactoring

### CODE REVIEW

File: binding.go [hunk 1]
```diff
@@ -24,6 +26,22 @@ type (
...
+	return defaultName
+}
+
+func (b *Binding) exchangeNameOr(defaultName string) string {
+	if b.ExchangeName != "" {
+		return b.ExchangeName
+	}
+
+	return defaultName
+}
```
Comment (MaxBreida) [inline binding.go:43]: why this naming? 😄 
fooNameOr

File: consume.go [hunk 2]
```diff
@@ -115,43 +119,45 @@ func (c *Consumer) startConsuming() error {
...
 	}
 
 	c.conn.logger.logDebug(fmt.Sprintf("Processing messages on %d message handlers", c.options.HandlerQuantity))
 
 	return nil
 }
 
-func (c *Consumer) handlerRoutine(deliveries <-chan amqp.Delivery, consumeOptions *ConsumeOptions, handler HandlerFunc) {
-	for msg := range deliveries {
+func (c *Consumer) handlerRoutine(deliveries <-chan amqp.Delivery, consumeOptions *ConsumeOptions) {
```
Comment (kevincali) [inline consume.go]: We could remove the parameter `consumeOptions` and use it from the receiver `c *Consumer` instead

File: binding.go (hunk 1 above)
Comment (MaxBreida) [reply binding.go:43]: could we give this another name? 😄

Comment (MaxBreida) [reply binding.go:43]: f it 🤷‍♂️

**Review by MaxBreida — APPROVED** [review]

**Pull request merged**